backend/clubs/utils.py

```diff
@@ -355,3 +362,49 @@ def get_django_minified_image(url, **kwargs):
     resp = requests.get(url)
     new_image = resize_image(resp.content, **kwargs)
     return ImageFile(io.BytesIO(new_image), name="image.png")
+
+
+class CacheManager:
```

Comment:
Do we need this class? Can we just monkeypatch the methods with exception handling/logging? 

Before we do this, can you also confirm that Django does not suppress cache connection exceptions automatically? I'm understanding that we're doing this because of the problems on the main page last night -- but let's confirm that this was because of the cache failing.

The logic in all 3 methods is mostly the same

Reply:
I'm fairly confident that a Redis connection refusal is fatal, and isn't suppressed by Django. This is reproducible by pointing the development cache to an incorrect Redis host, and accessing a club's information (`/api/clubs/<club_code>`). Attaching a screenshot below:

<img width="1440" alt="Screenshot 2024-02-11 at 4 29 46 PM" src="https://github.com/pennlabs/penn-clubs/assets/69180850/6193ec56-8778-464c-b35d-901655d86824">